src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/params.py

```diff
@@ -635,3 +637,24 @@ def _configure_security_policy_storage_params(cmd):
                options_list=('--end-ip-address',),
                help='The end IP address of the firewall rule. Must be IPv4 format. Use value'
                ' \'0.0.0.0\' to represent all Azure-internal IP addresses.')
+
+
+#####
+#           sql server vnet-rule
+#####
+
+with ParametersContext(command='sql server vnet-rule') as c:
+    # Help text needs to be specified because 'sql server vnet-rule create' is a custom
+    # command.
+    c.argument('server_name',
+               options_list=('--server', '-s'), completer=get_resource_name_completion_list('Microsoft.SQL/servers'))
+
+    c.argument('virtual_network_rule_name',
+               options_list=('--name', '-n'))
+
+    c.argument('virtual_network_subnet_id',
+               options_list=('--subnet'))
+
+register_extra_cli_argument('sql server vnet-rule', 'vnet-name', options_list=('--vnet-name'),
```

Comment:
@tjprescott , the register_extra_cli_argument call is not working for me. The vnet-name argument is not showing up. Ideas?

Reply:
Needs to be `sql server vnet-rule create`.  Registering extra CLI arguments must be done at a command level (not a subgroup level).